### PR TITLE
Bitfield functions and tests

### PR DESCRIPTION
**Pull request by jannikluhn**
See #11

- Added helper functions `has_voted`, `set_voted`, and `get_bitfield_length`
- Added tests for above
- Took the liberty to replace `^` (XOR) with `|` (OR) so that calling `set_voted` twice doesn't remove the vote
- Now waiting for PR #16 to be merged so that I can integrate the helpers to `state_transition.py`.

### CODE REVIEW

**Comment by jannikluhn** [comment]
- Replaced all bitfield calculations with the new helpers (hope I didn't miss any)
- Added another little helper `get_vote_count`
- Renamed bitmasks to bitfields and `signer_bitmask` to `notary_bitmask`

That should cover everything in #11, so it's ready for review.

**Comment by djrtwo** [comment]
You have a merge conflict. Do you mind resolving it?

**Comment by jannikluhn** [comment]
Rebased

File: tests/state/conftest.py [hunk 1]
```diff
@@ -265,11 +270,12 @@ def make_unfinished_block(parent_state,
             print('Making crosslink in shard %d' % shard)
             indices = get_crosslink_notaries(crystallized_state, shard, crosslink_shards=crosslink_shards, config=config)
             print('Indices: %r' % indices)
-            bitfield = [1 if random.random() < crosslinker_share else 0 for i in indices]
-            bitmask = bytearray((len(bitfield)+7) // 8)
-            for i, b in enumerate(bitfield):
-                bitmask[i//8] ^= (128 >> (i % 8)) * b
-            print('Bitmask:', bin(int.from_bytes(bitmask, 'big')))
+            is_voting = [random.random() < attester_share for _ in indices]
```
Comment (djrtwo) [inline tests/state/conftest.py]: Let's change this to `is_notarizing` and `notarizing` to distinguish it from the attestations above.

File: tests/utils/test_bitfield.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        (9, 2),
+        (16, 2),
+        (17, 3),
+    ]
+)
+def test_bitfield_length(attester_count, bitfield_length):
+    assert get_bitfield_length(attester_count) == bitfield_length
+
+
+
```
Comment (djrtwo) [inline tests/utils/test_bitfield.py]: only need 2 new lines here

File: tests/utils/test_bitfield.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+def test_empty_bitfield():
+    attesters = list(range(10))
+    bitfield = b'\x00' * get_bitfield_length(len(attesters))
```
Comment (djrtwo) [inline tests/utils/test_bitfield.py]: This should probably use `get_empty_bitfield`

File: tests/utils/test_bitfield.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+    for attester in attesters:
+        assert not has_voted(bitfield, attester)
+
+
+def test_bitfield_single_votes():
+    attesters = list(range(10))
+    bitfield = b'\x00' * get_bitfield_length(len(attesters))
```
Comment (djrtwo) [inline tests/utils/test_bitfield.py]: use `get_empty_bitfield`

File: tests/utils/test_bitfield.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+            if attester == voter:
+                assert has_voted(bitfield, attester)
+            else:
+                assert not has_voted(bitfield, attester)
+
+
+def test_bitfield_all_votes():
+    attesters = list(range(10))
+
+    bitfield = b'\x00' * get_bitfield_length(len(attesters))
```
Comment (djrtwo) [inline tests/utils/test_bitfield.py]: `get_empty_bitfield`

File: tests/utils/test_bitfield.py [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    for attester in attesters:
+        assert has_voted(bitfield, attester)
+    assert bitfield == b'\xff\xc0'
+
+
+def test_bitfield_some_votes():
+    attesters = list(range(10))
+    voters = [0, 4, 5, 9]
+
+    bitfield = b'\x00' * get_bitfield_length(len(attesters))
```
Comment (djrtwo) [inline tests/utils/test_bitfield.py]: `get_empty_bitfield`

File: beacon_chain/utils/bitfield.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
+def has_voted(bitfield, index):
```
[the remark this answers is not in the document]
Comment (jannikluhn) [reply beacon_chain/utils/bitfield.py]: Hm, that's a good point. Not sure either, so I'm fine with keeping it like this for now too.

File: tests/state/conftest.py (hunk 1 above)
Comment (jannikluhn) [reply tests/state/conftest.py]: Also renamed the remaining `is_voting` to `is_attesting` which seems to be the correct terminology.

**Comment by djrtwo** [comment]
lgtm

**Pull request merged**